src/Microsoft.Azure.WebJobs.Host/Queues/Bindings/QueueBindingProvider.cs

```diff
@@ -187,7 +187,7 @@ private IAsyncCollector<IStorageQueueMessage> BuildFromQueueAttribute(QueueAttri
 
             internal IStorageQueue GetQueue(QueueAttribute attrResolved)
             {
-                var account = Task.Run(() => this._accountProvider.GetStorageAccountAsync(attrResolved, CancellationToken.None)).GetAwaiter().GetResult();
```

Comment:
One issue with this is that this method is called extremely frequently - running all these tasks on the threadpool, most of which will be in a completed state, seems like it could be a negative.

Perhaps the answer is to move storage account validation out of the 'getaccount' method - it's the only reason it's an async call in the first place